Modules/Server/Sources/PocketCastsServer/Public/Cache/ShowInfoDataRetriever.swift

```diff
@@ -14,11 +18,30 @@ public actor ShowInfoDataRetriever {
         }
 
         let url = ServerHelper.asUrl(ServerConstants.Urls.cache() + "mobile/show_notes/full/\(podcastUuid)")
-        let request = URLRequest(url: url, cachePolicy: .reloadRevalidatingCacheData)
+        var request = URLRequest(url: url, cachePolicy: .reloadRevalidatingCacheData)
+
+        if let cachedResponse = cache.cachedResponse(for: request) {
```

Comment:
I expected `URLSession` to take care of this for us, but guess what: it doesn't.

I'm not sure if this is because the `mobile/show_notes/full/` redirects to another URL but the cache never worked for some podcasts (while it did work for some others).

Reply:
Could it because of the policy specified on the request: `.reloadRevalidatingCacheData` does our server works correctly with those?

Reply:
I also noticed that we use `URLSession.shared` for our requests, this has a cache but it's being shared with all other network request so maybe it can get easily recycled.